### PR TITLE
Add/block mover component readme

### PR DESCRIPTION
**Pull request by JustinyAhin**
## Description
Added documentation for the block mover component (see #22891)

## How has this been tested?
N/A, documentation only

## Types of changes
Documentation

## Checklist:
- [n/a] My code is tested.
- [n/a] My code follows the WordPress code style. 
- [n/a] My code follows the accessibility standards. 
- [n/a] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by JustinyAhin** [comment]
@youknowriad I was wondering where a contributor relatively new to some components (me 😀) can find docs about it.
For instance, for this particular one, I had to search through the code to find where some of the props are used; and guess what others do 😀.

**Comment by youknowriad** [comment]
These components in general are components that are not meant to be used by third-party developpers a lot, it's for advanced developpers that want to build their own editors. so right now, they are not really documented anywhere. In fact the work you're doing is the first attempt at this which is great.

File: packages/block-editor/src/components/block-mover/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#### isHidden
+
+#### rootClientId
+
+#### orientation
+
+The orientation of the block movers, vertical or horizontal.
+
+-   Type: `String`
+-   Required: `No`
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-mover/README.md:67]: I believe all the props aside `clientIds` are props provided by the interal withSelect which means you should remove them from the README.

File: packages/block-editor/src/components/block-mover/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Development guidelines
+
+### Usage
+
+Shows the block mover buttons in the block toolbar.
+
+```jsx
+import { BlockMover } from '@wordpress/block-editor';
+<BlockMover clientIds={ [ clientId ] } />
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-mover/README.md:22]: Can we do the same as BlockBreadcrumb here (wrapping?)

Comment (JustinyAhin) [reply packages/block-editor/src/components/block-mover/README.md]: Looks fine I think.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Pull request merged**